### PR TITLE
[pythonic resources] Update dagster-datahub to use Pythonic resources

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Updates the `dagster-datahub` package to offer Pythonic versions of the resources.
Shares config schema between both types of resources.

## Test Plan

New unit tests.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #14453** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14453" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/14453?utm_source=stack-comment).

File: python_modules/libraries/dagster-datahub/dagster_datahub/resources.py [hunk 1]
```diff
@@ -8,79 +11,82 @@
     KafkaEmitterConfig,
 )
 from datahub.emitter.rest_emitter import DatahubRestEmitter
+from pydantic import Field as PyField
```
Comment (schrockn) [inline python_modules/libraries/dagster-datahub/dagster_datahub/resources.py:14]: Let's not alias this to be consistent

File: python_modules/libraries/dagster-datahub/dagster_datahub/resources.py [hunk 2]
```diff
@@ -8,79 +11,82 @@
...
+    token: Optional[str] = PyField(default=None, description="Personal Access Token")
+    connect_timeout_sec: Optional[float] = PyField(default=None)
+    read_timeout_sec: Optional[float] = PyField(default=None)
+    retry_status_codes: Optional[List[int]] = PyField(default=None)
+    retry_methods: Optional[List[str]] = PyField(default=None)
+    retry_max_times: Optional[int] = PyField(default=None)
+    extra_headers: Optional[Dict[str, str]] = PyField(default=None)
+    ca_certificate_path: Optional[str] = PyField(default=None)
+    server_telemetry_id: Optional[str] = PyField(default=None)
+    disable_ssl_verification: bool = PyField(default=False)
```
Comment (schrockn) [inline python_modules/libraries/dagster-datahub/dagster_datahub/resources.py:28]: can just do `connect_timeout_sec: Optional[float] = None` for all of these, no?

**Review by schrockn — CHANGES_REQUESTED** [review]
Some nits

File: python_modules/libraries/dagster-datahub/dagster_datahub/resources.py (hunk 2 above)
Comment (benpankow) [reply python_modules/libraries/dagster-datahub/dagster_datahub/resources.py]: yup 👍

**Review by schrockn — APPROVED** [review]

**Pull request merged**